### PR TITLE
Allow server deletion

### PR DESCRIPTION
**Pull request by lbischof**

### CODE REVIEW

File: server/tower/shared.go [hunk 1]
```diff
@@ -107,23 +109,39 @@ func removeBlacklistedParameters(json *gabs.Container) *gabs.Container {
...
+			}
+		}
+		log.Printf("Job template %v allowed for %v", job_template, username)
+		return nil
 	}
 	return fmt.Errorf("Username %v tried to launch job template %v. Not in allowed job_templates", username, job_template)
 }
 
+func checkServicePermissions(template jobTemplatePermission, json *gabs.Container, username string) error {
+	if template.Validate == "metadata.uos_group" {
```
Comment (ganto) [inline server/tower/shared.go:135]: Where is this `Validate` attribute defined? I somehow cannot find any documentation or example for this.

I understand the intention of the function to be generic permission check, but somehow the implementation doesn't feel generic at all. If I have a job to define a "random" DNS record there might not be `uos_group` or `unifiedos_hostname` involved at all.

**Comment by ganto** [comment]
I generally feel the flow of the code is not super clear (to me). In my opinion the functions could definitely use a bit of comments to summarize the intention of the code.

File: server/tower/shared.go (hunk 1 above)
Comment (lbischof) [reply server/tower/shared.go:149]: I tried to add comments, so that everything is clear. Please tell me if something is still unclear.

**Comment by ganto** [comment]
Thanks a lot for the comments. Much easier to read now. lgtm

**Pull request merged**